### DISCUSSION TITLE
UUID package proposal

### DISCUSSION THREAD

**Discussion by cbr9** [espanso/espanso#1217] [Feature requests and Ideas]
So I was reading through [How I Capture Reading Notes in Obsidian](https://www.jamierubin.net/2021/02/15/how-i-capture-reading-notes-in-obsidian/) and the author mentions that he uses UUIDs to name highlights and notes from books. I also started using espanso yesterday (and it's honestly one of the best pieces of software I've tried in a long time, so congrats!!).

Naturally, I thought "hey, it would be really cool to be able to generate UUIDs from a espanso trigger like :id".

Would this be possible/easy to implement? I haven't peeked at the code yet

**Comment by cbr9** [comment]
Nevermind, I just saw you can create custom scripts. This is awesome.

**Comment by federico-terzi** [comment]
Hey @cbr9,

Thank you for the feedback! You can use scripts for this, but I think a custom variable extension would be handy (I use espanso for uuids myself!). So I'll mark it as a feature request, and we can start working on it once #594 is merged

Cheers :)
